### PR TITLE
daemon: change policyAdd message type from Info to Debug for dns policies

### PR DESCRIPTION
**Pull request by aanm**
Also, change the way we print the rules by also printing all fields
names of the policy being imported.

Fixes: b0aeb2888ac6 ("policy/api: add String function for Rules type")
Signed-off-by: André Martins <andre@cilium.io>

---
This change is [Reviewable](https://reviewable.io/reviews/cilium/cilium/6019)

### CODE REVIEW

**Comment by aanm** [comment]
test-me-please

**Comment by tgraf** [comment]
The info message was quite useful in the past. What is the motivation to turn it into a Debug?

**Comment by aanm** [comment]
@tgraf it's extremely verbose when importing a rule that contains a DNS rule as the DNS poller is calling policyAdd every 5 seconds.

**Comment by tgraf** [comment]
@aanm Even if the poller is returning a constant list of IPs? Is it the random ordering or constantly changing subset of IPs that makes it verbose?

**Comment by ianvernon** [comment]
Being able to see what rules we are adding to the repository is crucial to know, even at Info level. Do we have any change detection logic here that we could leverage? I.e., only log when rules are changed at Info level?

**Comment by aanm** [comment]
> hat makes it verbose?

@raybejjani would be better suitable to answer that question than me

**Comment by joestringer** [comment]
As mentioned in the SIG-Policy meeting, there should be a "generated" flag for DNS-generated CIDR policy insertions which may provide an intermediate fix for the verbosity issue.

**Comment by aanm** [comment]
test-me-please

**Review by ianvernon — APPROVED** [review]

File: daemon/policy.go [hunk 1]
```diff
@@ -161,6 +161,9 @@ type AddOptions struct {
 	// ReplaceWithLabels if present indicates that existing rules with the
 	// given LabelArray should be deleted.
 	ReplaceWithLabels labels.LabelArray
+	// Generated should be set as true to signalize a the policy being inserted
+	// was generated by cilium-agent, e.g. dns pooler.
```
Comment (tgraf) [inline daemon/policy.go]: poller

**Review by tgraf — CHANGES_REQUESTED** [review]

**Comment by aanm** [comment]
test-me-please

**Review by tgraf — APPROVED** [review]

**Pull request merged**